### PR TITLE
Add raw incidents metrics

### PR DESCRIPTION
**Pull request by MaggotMouth**
Implement endpoints for single and multiple raw incident metrics

https://developer.pagerduty.com/api-reference/b3A6Mjc0ODExMw-get-raw-data-single-incident

https://developer.pagerduty.com/api-reference/c2d493e995071-get-raw-data-multiple-incidents

This was previously done in PRs #347 and #422 which have gone stale.

### CODE REVIEW

**Comment by krav** [comment]
@ChuckCrawford @theckman Hi! Can we get this reviewed? Been looking to use this endpoint. Thanks!

File: analytics.go [hunk 1]
```diff
@@ -94,3 +151,45 @@ func (c *Client) getAggregatedData(ctx context.Context, analytics AnalyticsReque
 
 	return analyticsResponse, nil
 }
+
+// GetRawDataSingleIncident gets the raw analytics for the requested incident.
+func (c *Client) GetRawDataSingleIncident(ctx context.Context, id string) (RawData, error) {
+	h := map[string]string{
+		"X-EARLY-ACCESS": "analytics-v2",
```
Comment (ChuckCrawford) [inline analytics.go:158]: They probably were at the time you put up the PR but these APIs are no longer "Early Access"

File: analytics.go [hunk 2]
```diff
@@ -94,3 +151,45 @@ func (c *Client) getAggregatedData(ctx context.Context, analytics AnalyticsReque
 
 	return analyticsResponse, nil
 }
+
+// GetRawDataSingleIncident gets the raw analytics for the requested incident.
+func (c *Client) GetRawDataSingleIncident(ctx context.Context, id string) (RawData, error) {
```
Comment (ChuckCrawford) [inline analytics.go:156]: Could we update the signatures to return a pointer to the result? e.g. `func (c *Client) GetRawDataSingleIncident(ctx context.Context, id string) (*RawData, error)`

Comment (ChuckCrawford) [inline analytics.go:156]: A few nitpicky comments about the function names: could we name these two functions after their [OpenAPI Operation Ids](https://github.com/PagerDuty/api-schema/blob/55f11455b56cd1a829e0a949ebba744087c75172/reference/REST/openapiv3.json#L18401): `getAnalyticsIncidentsById` and `getAnalyticsIncidents`?

File: analytics.go [hunk 3]
```diff
@@ -61,6 +88,36 @@ type AnalyticsData struct {
 	RangeStart                     string  `json:"range_start,omitempty"`
 }
 
+// RawData represents the structure of the raw analytics we have available.
+type RawData struct {
```
Comment (ChuckCrawford) [inline analytics.go:92]: Could we name this `AnalyticsRawIncident` ?

File: analytics.go [hunk 4]
```diff
@@ -23,6 +37,19 @@ type AnalyticsResponse struct {
 	TimeZone      string           `json:"time_zone,omitempty"`
 }
 
+// RawDataResponse represents the response from the PagerDuty API.
+type RawDataResponse struct {
```
Comment (ChuckCrawford) [inline analytics.go:41]: This would be `AnalyticsRawIncidentsResponse`

File: analytics.go [hunk 5]
```diff
@@ -15,6 +17,18 @@ type AnalyticsRequest struct {
 	TimeZone      string           `json:"time_zone,omitempty"`
 }
 
+// RawDataRequest represents the request to be sent to PagerDuty when you want
+// raw analytics.
+type RawDataRequest struct {
```
Comment (ChuckCrawford) [inline analytics.go:22]: This would be `AnalyticsRawIncidentsRequest`

**Review by ChuckCrawford — CHANGES_REQUESTED** [review]
Thank you for the PR! If would wouldn't mind making a few changes we can get this merged...

**Comment by MaggotMouth** [comment]
Thanks for the review @ChuckCrawford - I've made the changes as requested and updated the associated tests.

**Review by ChuckCrawford — APPROVED** [review]
Thank you for your contribution @MaggotMouth.

**Comment by Tyrael** [comment]
@ChuckCrawford is there any ETA for v1.9.0? this was merged more than a year ago but there were no releases since.

**Pull request merged**